### ISSUE TITLE
[plot] scatter profile tests failure

### ISSUE THREAD

**Issue by t20100** [silx-kit/silx#1909]
This error happened on appveyor with Python 3.6, PyQt5:
```
test (silx.gui.plot.tools.test.testScatterProfileToolBar.TestScatterProfileToolBar)
Test ScatterProfileToolBar ... QThread: Destroyed while thread is still running
```

**Comment by t20100** [comment]
This seems random

**Comment by vallsv** [comment]
I think it is because your thread is own by the widget.

If the widget is destroyed before the thread have finished, you will have this issue.

Can be reproduced with the following code (which segfault on Windows)
```
from PyQt5 import Qt

app = Qt.QApplication([])

class TestThread(Qt.QThread):

    def run(self):
        self.sleep(1000)

w = Qt.QWidget()
t = Qt.QThread(w)
t.start()
w.deleteLater()

app.exec_()
```

**Comment by vallsv** [comment]
Could be fixed on the tests by waiting for finish of pending actions (i usually create a `hasPendingOperations` on the widget). Or kill the thread when you delete your widget.

**Comment by t20100** [comment]
Calling `terminate` on the thread while it is computing a Delaunay leads to errors when starting another Delaunay afterwards...
`hasPendingOperations` would be usefull for the tests, but the issue in real application remains.

**Comment by t20100** [comment]
closed by #1912
